core/modules/widgets/checkbox.js

```diff
@@ -212,6 +212,10 @@ CheckboxWidget.prototype.handleChangeEvent = function(event) {
 		} else {
 			listContents = $tw.utils.parseStringArray(this.wiki.extractTiddlerDataItem(this.checkboxTitle,this.checkboxListIndex) || "") || [];
 		}
+		if(Object.isFrozen(listContents)) {
```

Comment:
I wonder if this is the best fix.

Back in line 211, in the codepath for `if(this.checkboxListField)` being true, the value is retrieved with `tiddler.getFieldList()`, which always returns an immutable value. Meanwhile, in the other codepath, `wiki.extractTiddlerDataItem()` also reuses results from earlier calls thanks to the caching mechanism (although there's no freezing, so the immutability is not enforced).

As a result, we cannot safely mutate either of the returned lists, so it would be better to do the cloning without the `isFrozen()` test.

Reply:
If I clone the list every time, then in any usage scenario where a custom list is used (which I expect to be most of the time), the clone will be unnecessary since parseStringArray creates a new list. I think a better solution will be to extract the field without calling `getFieldList`, then examine the field I get. If it's a list, clone it before modifying. If it's a string, use `parseStringArray` and then don't clone the results as we know that that's a brand-new list.

I'll implement that sometime soon (not right now, as I'm about to run out the door for an appointment).

Reply:
Thanks for your patience @rmunn – your proposed solution makes sense, and I think would be preferable.

Reply:
Sorry it took a week for me to get to this; I've been traveling without good Internet access until now. I've implemented that suggestion (just make a defensive copy if the field was already a list, whether or not it was frozen). I also fixed another bug mentioned in #6871, where a non-existent tiddler would result in `tiddler is null` in this same piece of logic.

There may be one more check I should make, which is to see if the field contains a data type (such as a date) that should not be turned into a list. (See https://github.com/Jermolene/TiddlyWiki5/issues/6871#issuecomment-1211578823 for discussion of this issue). I'm running out of time to implement that today, so I'll do that tomorrow. Then the PR will be ready for review again.